src/app/state/account/selectors.ts

```diff
@@ -12,3 +13,44 @@ export const selectAccountAddress = createSelector([selectAccount], account => a
 export const selectAccountAvailableBalance = createSelector([selectAccount], account => account.available)
 export const selectAccountIsLoading = createSelector([selectAccount], account => account.loading)
 export const selectAccountAllowances = createSelector([selectAccount], account => account.allowances)
+export const selectPendingTransactions = createSelector(
+  [selectAccount],
+  account => account.pendingTransactions,
+)
+export const selectAccountNonce = createSelector([selectAccount], account => account.nonce ?? '0')
+export const selectPendingTransactionForAccount = createSelector(
+  [selectPendingTransactions, selectSelectedNetwork],
+  (pendingTransactions, networkType) => pendingTransactions[networkType] ?? [],
+)
+export const hasAccountUnknownPendingTransactions = createSelector(
+  [selectAccountNonce, selectTransactions, selectAccountAddress],
+  (accountNonce, transactions, accountAddress) => {
+    // TODO: This logic fails, in case transaction are not from the initial page of account's transactions
+    const filteredTransactions = transactions.filter(({ from }) => from && from === accountAddress)
+
+    if (filteredTransactions.length === 0 && BigInt(accountNonce) === 0n) return false
+
+    const maxNonceFromTxs = filteredTransactions.reduce(
+      (acc, { nonce }) => {
+        if (!nonce) {
+          return acc
+        }
+
+        const bigNonce = BigInt(nonce)
+
+        if (!acc) {
+          return bigNonce
+        }
+
+        return bigNonce > acc ? bigNonce : acc
+      },
+      undefined as bigint | undefined,
+    )
+
+    if (!maxNonceFromTxs) {
+      return BigInt(accountNonce) > 0n
+    }
+
+    return BigInt(maxNonceFromTxs) + BigInt(1) < BigInt(accountNonce)
```

Comment:
https://7ad67e19.oasis-wallet.pages.dev/account/oasis1qpw4ryz88wfe9mmy2sqpkevlzp4ua8ec7udxfqts
incorrectly always shows "Some transactions are currently in a pending state"

Reply:
Nice catch! Fixed here https://github.com/oasisprotocol/oasis-wallet-web/pull/1954/commits/5e2036b61d2f0e070dc60a51ff34f2ff7fa872bf the issue was that 0 nonce was not correctly parsed from tx.